### ISSUE TITLE
Passing env or dynamic values to config.json

### ISSUE THREAD

**Issue by yasirmturk** [apollographql/apollo-ios#2707]
## Feature request

It is not possible to pass a value to config file. 

## Motivation

A very common example is the auth header. The token is generate on the runtime using script or saved in env variable and it is not possible to plug in that into the config.json file.

## Proposed solution

Use placeholders in config.json for env variables

## Outstanding Questions

```
"headers": [
            {
                "key": "Authorization",
                "value" : "Bearer $(authToken)"
            }
        ],
```

**Comment by calvincestari** [comment]
Thanks for the suggestion @yasirmturk, this would be a pretty useful feature.

Currently the JSON is simply passed into `JSONDecoder` so we'd need to have a first pass over the JSON to fetch and replace any environment variables before handing it off to `JSONDecoder`. Or something like that..

**Comment by anashkevichp** [comment]
is there any update on @calvincestari?
If that's still in progress, maybe you could suggest any workaround to pass actual API token into <headers> value. Is there anything better rather that script modifying .json configuration file on the fly?

**Comment by calvincestari** [comment]
Hi @anashkevichp. It's in progress as much as the feature request exists but it's not in scheduled in anyone's upcoming work to be done.

The only workaround is performing codegen in code (Swift) instead of the JSON + CLI combination. Modifying JSON on the fly would be tedious.

**Comment by AnthonyMDev** [comment]
Yeah, for the foreseeable future, I think the best way to handle this would be to [use the Swift Codegen library](https://www.apollographql.com/docs/ios/code-generation/run-codegen-in-swift-code), as @calvincestari recommended. For advanced use cases that can't be handled by the CLI, it offers a lot more flexibility.

**Comment by anashkevichp** [comment]
thanks guys, that could work for me!
However, this message that `ApolloCodegenLib` limited only for macOS really confuses me ([here](https://www.apollographql.com/docs/ios/code-generation/run-codegen-in-swift-code/)). Is it safe to use for iOS projects?
<img width="688" alt="Screenshot 2024-01-24 at 5 43 45 PM" src="https://github.com/apollographql/apollo-ios/assets/1592771/7dc71dca-4910-4942-8c1e-b18e549b2fb1">

**Comment by calvincestari** [comment]
Code generation is a development workflow so there is no need to for it to be able to run on iOS; it's not something you ship with your app. You'll need to add another target that can run on macOS only.

**Comment by yasirmturk** [comment]
is there a documentation about how to deploy it as part of CI/CD and use code gen without integrating it in the app source project? 🤔

**Comment by calvincestari** [comment]
We do not have any documentation to explain that workflow. Our typical recommendation has been to include it with the app source but it's a target that's not distributed.
